### PR TITLE
update result analyze script based on Ranran's fix

### PR DESCRIPTION
**Pull request by zpcore**

### CODE REVIEW

File: benchmarks/util.py [hunk 1]
```diff
@@ -141,14 +141,10 @@ def get_cpu_name():
 
 
 def get_gpu_name():
-  gpu_names = subprocess.check_output(
+  return subprocess.check_output(
```
Comment (Liyang90) [inline benchmarks/util.py:144]: Is this change intended or not? It seems to be reverting to older codes.

File: benchmarks/result_analyzer.py [hunk 2]
```diff
@@ -43,7 +54,6 @@ def run(self):
         "accelerator": pd.Series(dtype="str"),
         "accelerator_model": pd.Series(dtype="str"),
         "xla": pd.Series(dtype="str"),
```
Comment (Liyang90) [inline benchmarks/result_analyzer.py:56]: "xla_flags" was removed here. Is it intentional or accidental reverting?

File: benchmarks/util.py (hunk 1 above)
Comment (zpcore) [reply benchmarks/util.py:144]: You are right, thanks, I have reverted this line back.

File: benchmarks/result_analyzer.py (hunk 2 above)
Comment (zpcore) [reply benchmarks/result_analyzer.py]: Thanks for catching that. I have added this line back.

**Comment by zpcore** [comment]
Add @RissyRan in this PR since she is the main author for the change.

File: benchmarks/result_analyzer.py [hunk 3]
```diff
@@ -58,13 +69,92 @@ def run(self):
...
     for file in jsonl_files:
-      metric_df = self.extract_metrics(file, metric_df)
+      metric_df = self.extract_metrics_csv(file, metric_df)
 
     # additional processing of the metric_df can be done here
 
     self.export_metric_report(metric_df)
 
-  def extract_metrics(self, file, metric_df):
+  def get_calculated_metrics(self, d, tmp):
```
Comment (Liyang90) [inline benchmarks/result_analyzer.py:78]: Could this method args be made more readable? It's okay to use `tmp` in the method call, but in method definition, it's better to reflect the actual object that is passed.

In this case `tmp` is a dict read from a json string representing the raw results of one experiment.

**Review by Liyang90 — COMMENTED** [review]
LGTM with minor suggestion

**Comment by miladm** [comment]
would be great to merge this PR today if at all possible. thank you @zpcore

**Comment by frgossen** [comment]
Just a note, we should merge this into the main branch

File: benchmarks/result_analyzer.py (hunk 3 above)
Comment (zpcore) [reply benchmarks/result_analyzer.py:78]: Thanks, I will make the update. By the way, we will drop this PR and instead, merge this into main branch directly.

**Review by frgossen — CHANGES_REQUESTED** [review]
This should be merged into the main branch

**Comment by zpcore** [comment]
Created a new PR: https://github.com/pytorch/xla/pull/5920 to replace this one.

**Comment by frgossen** [comment]
Thanks!
